gateway_to_backend/README.md

```diff
@@ -0,0 +1,591 @@
+to avoid additional constraints on the gateway implementation. In fact,
+it implies the generation of a PING request to the broker if there is no
+message generated within a keep alive interval.
+
+Nevertheless, it is recommended to use the keep alive mechanism of MQTT
+to overcome the half-open connection issue of a TCP connection.
+
+The broker will disconnect a client if no message is received within one
+and a half times of the keep alive interval and the last will message
+will be sent. It allows the backends to be notified in a really short
+period of a problem with a gateway connection.
+
+60s is a good value for the keep alive interval.
+
+### Config module
+
+---
+
+This module oversees configuration of the different sinks attached to
+the gateway.
+
+#### Get configs message
+
+- **Request:**
+
+    > **topic:** gw-request/get_configs/*\<gw-id\>*
+    >
+    > **content:** [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[GetConfigsReq][message_GetConfigsReq]
+
+- **Response:**
+
+    > **topics:** gw-response/get_configs/*\<gw-id\>*
+    >
+    > **content:** [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[GetConfigsResp][message_GetConfigsResp]
+    >              and [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[SinkReadConfig][message_SinkReadConfig]
+
+:warning:
+
+_A gateway must generate a [GetConfigsResp][message_GetConfigsResp]
+with a request id set to 0 in its header when its configuration change.
+When a sink is added or removed for example.
+It is the only way for a backend to monitor this change without
+having to frequently poll for such event._
+
+#### Set config message
+
+-  **Request:**
+
+    > **topic:** gw-request/set_config/*\<gw-id\>/\<sink-id\>*
+    >
+    > **content:** [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[SetConfigReq ][message_SetConfigReq] and [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[SinkNewConfig][message_SinkNewConfig]
+
+- **Response:**
+
+    > **topics:** gw-response/set_config/*\<gw-id\>/\<sink-id\>*
+    >
+    > **content:** [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[SetConfigResp][message_SetConfigResp]
+
+:warning:
+
+_A gateway must generate a [SetConfigResp][message_SetConfigResp]
+with a request id set to 0 in its header when a sink state
+change, especially if it reboots. In fact, if sink configuration
+is changed through remote api, it is the only way for a backend
+to monitor this change._
+
+#### Get gateway info
+
+- **Request:**
+
+    > **topic:** gw-request/get_gateway_info/*\<gw-id\>*
+    >
+    > **content:** [GenericMessage][message_GenericMessage].[WirepasMessage][message_WirepasMessage].[GetGwInfoReq][message_GetGwInfoReq]
+
+- **Response:**
+
+    > **topics:** gw-response/set_config/*\<gw-id\>/\<sink-id\>*
```

Comment:
it is not set_config but get_gateway_info.
It was already wrong in pdf version of wp-rm-128...

Reply:
ok, i'll fix it